src/lib/pages/backoffice/Actions/CheckIn/ItemsSearch/ItemsSearch.js

```diff
@@ -16,13 +17,35 @@ export default class ItemsSearch extends Component {
       queryString: propsQueryString,
       checkin,
       clearSearchBar,
+      isCheckout,
     } = this.props;
     queryString = queryString || propsQueryString;
     if (queryString.trim() === '') return;
-    await checkin(queryString);
+    isCheckout
+      ? await this.performCheckout(queryString)
+      : await checkin(queryString);
     clearSearchBar();
   };
 
+  performCheckout(queryString) {
+    const { checkoutData, loan, performCheckoutAction } = this.props;
+    checkoutData.hits.forEach(hit => {
+      if (hit.metadata.barcode === queryString) {
+        this.props.loadingHandler();
+        const checkoutUrl = loan.availableActions.checkout;
+        const patronPid = loan.metadata.patron_pid;
+        const documentPid = hit.metadata.document.pid;
```

Comment:
Just to be on the safe side:
```suggestion
        const documentPid = hit.metadata.document_pid;
```
* `document_pid` is a real field of the loan
* `document` is a `$ref`, resolved by the backend
I would prefer the first whenever possible